eox_nelp/user_profile/api/v1/views.py

```diff
@@ -56,14 +57,17 @@ def update_user_data(request):
             # This extra code block is required since the method update_account_settings just
             # allows to update fields defined in the AccountUserSerializer and the AccountLegacyProfileSerializer
             # so some fields like first_name and last_name are not editable in the standad implementation.
-            extra_account_user_fields = getattr(settings, "EXTRA_ACCOUNT_USER_FIELDS", [])
+            # so some fields related ExtraInfo are not editable in the standad implementation.
 
-            if extra_account_user_fields:
-                for field, value in request.data.items():
-                    if field in extra_account_user_fields and hasattr(request.user, field):
-                        setattr(request.user, field, value)
+            extra_account_user_fields = getattr(settings, "EXTRA_ACCOUNT_USER_FIELDS", [])
+            required_user_extra_info_fields = getattr(settings, 'REQUIRED_USER_EXTRA_INFO_FIELDS', [])
 
-                request.user.save()
+            for field, value in request.data.items():
```

Comment:
if the request.data.items() has 10000 records this will affect the API performance since you remove the previous conditional, indeed could you change the implementation to use a for something like this

```python
            for field in extra_account_user_fields:
                if hasattr(request.user, field):
                    setattr(request.user, field, value)
               request.user.save()

            for field in required_user_extra_info_fields:
                save_extrainfo_field(request.user, field, value)
```